### ISSUE TITLE
PUT & DELETE Issue

### ISSUE THREAD

**Issue by iainmcfall** [mevdschee/php-crud-api#543]
I have an issue where the GET & POST works but PUT & DELETE don't.

I'm running on:
- IIS v10
- mySQL v8.0.13
- PHP 7.2.7

Using Postman:
GET records/person    _> returns the list of every record in the person table_
GET records/person/1    _> returns the record with ID = 1 from person table_
POST records/person      _> returns the ID of the newly created record in person table_
        {name: "Chris"}
PUT records/person/1       _**> returns a 404 error**_
        {name: "Alex"}
DELETE records/person/1       _**> returns a 404 error**_

I updated my IIS configuration to accept PUT & DELETE (via Handler Mappings) but still get a 404

I then tried the workaround I found in issue **https://github.com/mevdschee/php-crud-api/issues/351**

Updated the api.php config with:    **'method'=>strtoupper($_GET['method']),**

Then I added **?method=put** to the URL in Postman.

Now I get a **500 internal server error** returned on all requests (GET, POST, PUT & DELETE) - which for some reason is being caused by the line added to the api.php config above.

What am I missing here?

**Comment by mevdschee** [comment]
Maybe this helps? See:

https://forums.iis.net/t/1166025.aspx

**Comment by iainmcfall** [comment]
Thanks. The lin kyou provided is related to a 405 error. I'm getting a 404 error (though may be the same symptom) Idk.

These fix suggestions are mainly related to WebDAV being installed - the fix being to remove it.-  and including the PUT & DELETE verbs in the <handlers> section of the web.config.

 I don't have WebDAV installed and my web.config already has the verbs in the <handlers> section.

I have tried a few other more recent suggestion I have found (again all mostly related to ensuring the server accepts PUT&DELETE but to no avail).

Before I un/reinstall IIS and start afresh - is there no workaround? It seems the config 'method' is no longer supported since the workaround you provided in issue 351 (which is why I presume I'm getting the 500 error when I try you r workaround).
